pkg/chunked/storage_linux_test.go

```diff
@@ -0,0 +1,171 @@
+package chunked
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// Mock for ImageSourceSeekable
+type mockImageSource struct {
+	streams chan io.ReadCloser
+	errors  chan error
+}
+
+func (m *mockImageSource) GetBlobAt(chunks []ImageSourceChunk) (chan io.ReadCloser, chan error, error) {
+	return m.streams, m.errors, nil
+}
+
+type mockReadCloser struct {
+	reader io.Reader
+	closed bool
+}
+
+func (m *mockReadCloser) Read(p []byte) (int, error) {
+	return m.reader.Read(p)
+}
+
+func (m *mockReadCloser) Close() error {
+	m.closed = true
+	return nil
+}
+
+func mockReadCloserFromContent(content string) *mockReadCloser {
+	return &mockReadCloser{reader: bytes.NewBufferString(content), closed: false}
+}
+
+func TestGetBlobAtNormalOperation(t *testing.T) {
+	errors := make(chan error, 1)
+	expectedStreams := []string{"stream1", "stream2"}
+	streamsObjs := []*mockReadCloser{
+		mockReadCloserFromContent(expectedStreams[0]),
+		mockReadCloserFromContent(expectedStreams[1]),
+	}
+	streams := make(chan io.ReadCloser, len(streamsObjs))
+
+	for _, s := range streamsObjs {
+		streams <- s
+	}
+	close(streams)
+	close(errors)
+
+	is := &mockImageSource{streams: streams, errors: errors}
+
+	resultChan, err := getBlobAt(is, nil)
+	require.NoError(t, err)
+
+	i := 0
+	for result := range resultChan {
+		assert.NoError(t, result.err)
+		buf := new(bytes.Buffer)
+		_, _ = buf.ReadFrom(result.stream)
+		result.stream.Close()
+		assert.Equal(t, expectedStreams[i], buf.String())
+		i++
+	}
+	assert.Equal(t, len(expectedStreams), i)
```

Comment:
Absolutely non-blocking, this works perfectly fine:
- `assert.Len` exists
- Alternatively, the loop could collect a `returnedContents [][]byte`, and afterwards there could be a single `assert.Equal(t, expectedStreams, returnedContents)` comparing both the length and data.